Gemfile

```diff
@@ -16,7 +16,7 @@ source 'https://rubygems.org'
 
 group :test do
   gem 'rake'
-  gem 'puppet', ENV['PUPPET_VERSION'] || '~> 3.8.0'
+  gem 'puppet', ENV['PUPPET_VERSION'] || '~> 4.0'
```

Comment:
@jyaworski This PR is proposing a major version update, and uses Puppet 4 language features. Therefore it is not compatible with any version of Puppet <4.0. I wrote a little more about that [here](https://github.com/voxpupuli/puppet-dotnet/pull/9#issuecomment-142127835).